### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by zeff96**
- [ ] Installed CanCanCan in my project.

- [ ] Added a role column to the users table.

- A user can delete a post if it is theirs or if they have an admin role.

- A user can delete a comment if it is theirs or if they have an admin role.

### CODE REVIEW

File: app/views/partials/_comments.html.erb [hunk 1]
```diff
@@ -1,4 +1,15 @@
...
+  </div>
+  <div>
+    <% if can? :destroy, comment %>
+      <%= link_to 'delete', [current_user, @post, comment], data: {
+        turbo_method: :delete,
+        turbo_confirm: 'Are you sure!'
+        }, class: 'btn btn-danger btn-sm' 
+      %>
+    <% end %>
+  </div>
```
Comment (andersonlebon) [inline app/views/partials/_comments.html.erb]: - [x] Kindly fix this error which is due to the wrong path put in the link_to tag
image

File: app/views/posts/show.html.erb [hunk 2]
```diff
@@ -6,7 +6,7 @@
   </p>
   <p><%= @post.text %></p>
   <div class="align-self-end">
-    <%= form_with url: post_likes_path(@post), method: :post do |f| %>
+    <%= form_with url: user_post_likes_path(current_user, @post), method: :post do |f| %>
```
Comment (andersonlebon) [inline app/views/posts/show.html.erb]: - [ ] Another error is happening when I wants to like a post,  the post is nil
image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @zeff96 && @shasherazi 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is now professional :heavy_check_mark: 
- Nice UI 👏  ✔️ 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @zeff96 and @shasherazi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ Fixed path errors requested by previous reviewer to ensure app is working without errors.
✔️ Professional written README file with all sections required.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by zeff96** [comment]
@danmainah, Thanks for the approval

**Pull request merged**